app/controllers/subscriptions_controller.rb

```diff
@@ -5,19 +5,21 @@ class SubscriptionsController < ApplicationController
 
   def create
     @subscription = current_user.subscriptions.build(conference_id: @conference.id)
-    if @subscription.save!
+    save_succeed = @subscription.save
+    if save_succeed
       redirect_to root_path, notice: "You have been subscribed to receive email notifications for #{@conference.short_title}."
     else
-      redirect_to root_path, error: subscription.errors.full_messages.to_sentence
+      redirect_to root_path, error: @subscription.errors.full_messages.to_sentence
     end
   end
 
   def destroy
     @subscription = current_user.subscriptions.find_by(conference_id: @conference.id)
-    if @subscription.destroy
+    if @subscription && @subscription.destroy
       redirect_to root_path, notice: "You have been unsubscribed and now you will not be receiving email notifications for #{@conference.short_title}."
     else
-      redirect_to root_path, error: @subscription.errors.full_messages.to_sentence
+      flash[:error] = 'You are not subscribed to that conference.'
```

Comment:
But what should happen if the destroy fail again even the subscription (if this is possible) ? 
if !@subscription
  # go to error
if @subscription.destoy
  # go to root_path
else 
  # does this else have to exist here?